### DISCUSSION TITLE
Is there a way to run this in strict mode...

### DISCUSSION THREAD

**Discussion by braindeaf** [openapi-ts/openapi-typescript#1811] [General]
I am coming to `openapi-typescript` from a Ruby background. We use RSwag to verify our API responses against our swagger json. In effect we can say the API response is correct and ensure the schema accounts for all responses from our API call, optional fields and all. This done with Ruby's json-schema gem. 

As the last part we use `openapi-typescript` to generate Typescript from our swagger.json this is all good well, but it would appear RSwag is configured to be strict to require all defined properties by default unless less otherwise configured not to and we have make them otherwise optional if necessary, The Typescript generated from our swagger file makes all properties optional by default but there doesn't seem to be a way to override this?

Am I missing something?

Many thanks

RobL

**Comment by drwpow** [comment]
I haven’t used RSwag, but Swagger, OpenAPI, and JSONSchema (OpenAPI being an extension of JSONSchema since 3.0) all specify properties being optional by default ([source](https://swagger.io/specification/)):

> In the following description, if a field is not explicitly REQUIRED or described with a MUST or SHALL, it can be considered OPTIONAL.

This is why both in schema objects as well as request bodies the `required` field is necessary to mark something as required.

This library doesn’t try and assert any opinions over how it “should” be; it just tries to follow the specification as closely as possible, and that is how the specification is outlined. To make something required, just add `required` where needed.

**Reply by braindeaf** [reply]
Excellent. How unusual that they should add this option at all, it seems quite out of place then.

https://www.rubydoc.info/gems/json-schema/2.4.1#:~:text=With%20the%20%3Astrict%20option%2C%20validation,treated%20as%20required%20by%20default.

I will look at adjusting our approach. Thank you.
